feat: resolve address to ENS

sasicodes opened the pull request:
- Create `useENS` hook to resolve user address to ENS service( avatar and name)
- No external lib installed

![image](https://user-images.githubusercontent.com/29498872/145669027-7681b87c-9b86-4fde-8af3-1b74e0ed71a4.png)

sasicodes committed:
feat: resolve address to ENS

sasicodes committed:
feat: validations and error handling

sasicodes committed:
fix: typos

appleseed-iii committed:
Alpha pr for Migration (#869)

* sync

* create v1Stake and C2A components

* some styling etc

* Visuals done

* styling

* moved it out of staking page

* call to action showing if old assets are detected

* detecting wallet balances and showing v1 stake or v2 stake conditionally

* placeholder link for now for learn more btn

* modal opening and closing

* theme works

* adding migrate button to v1 staking page

* help text added

* Checkpoint

* UI reactive to allowance state

* input exists for unstakers

* Adapt to migrate-all

* small css

* Fix call

* Minor fixes and change approval calls

* Remove unnecessary stuff

* Add contract abis

* Add new contracts

* Fix error

* Remove typechain interfaces and add generator command

* Ignrore src/typechain

* Delete typechain

* approve button dispatch waiting

* remove async await

* Migration functionality

* new addresses for testnet contracts added

* new addies

* remove typo

* new contracts in v2 stake

* comments added

* css changes to call to action alert

* detecting ohm v2 balance correctly

* more pulling from v2 contracts

* Update addresses

* Revert ohm v2

* approving works in v2

* V2 migration address updates (#690)

* Update addresses

* Revert ohm v2

* remove approve button from v1 stake

* fix broken error

* corrected addresses back

* Improved error handling

* Migration fully working

* Quick fix

* unstaking works for old sOhm

* Add tooltips

* Alignment and sizing

* Fix learn more link

* Move allowance load call

* Fix addresses

* added abi

* Empty commit

* removr tabs add second dropdown

* almost working

* errors but I think its right

* Fix approval bug

* Insert estimated quantity

* Fix style

* Fix errors

* Change networkId

* Fix avax

* Integrate avax migrate functionality

* Trigger CICD

* Fix gohm approval units

* Fix approval checks

* Revert styling

* Remove logging

* Add pending transactions

* Fix styling

* Fix gohm unwrap permissions

* slight tweak to select styles and added to gohm on avax button

* changed name to remove warning

* Fix errors in console

* Fix loadaccount details

* Revert unnecessary changes

* Fix errors

* Fix addresses

* Update text

* Add v2 handling for staking

* Add approval changing

* Insert v2

* Remove wsOHM option from wrap

* Various fixes

* Fix name

* Fix -1 networkId bug

* Remove logging

* fixed learn more links

* UI adjustments

* change back

* centered & spaced learn more & migrate btns

* change icon

* alignment in migration modal updated

* symetrical

* added more detailed balances to v1 stake

* toggle back to v1 added

* fix migration banner layout

* v1 v2 toggle added

* important typo

* added v1 symbol to clarify on v1 stake page

* fixed modal overly transparency

* migration modal is useable in mobile

* v1 v2 toggle only appears if user has both kinds of assets

* users are forced to claim outstanding v1 bonds befor emigrating

* c2a doest show with outstanding bonds

* push

* activebonds

* wsohm patch

* change back for debugging

* tiny type fix

* fixing modal

* fixing view to display all assets whether they are owned or not

* typo

* show v2 balances on v2 stake page

* wsOHM balance fix

* dont show 0 balances

* wrap page & prettier

* mainnet

* update wrap allowance check

* note

Co-authored-by: Girth Brooks <girth.eth@gmail.com>
Co-authored-by: borbersaur <plasmabeach@gmail.com>
Co-authored-by: mrunbanked <mrunbanked@protonmail.com>
Co-authored-by: mrunbanked <92614968+mrunbanked@users.noreply.github.com>
Co-authored-by: borbesaur <87902542+borbesaur@users.noreply.github.com>
Co-authored-by: casuwu <taken404@gmail.com>

appleseed-iii committed:
Revert "Alpha pr for Migration (#869)" (#873)

This reverts commit ace6e834baf83e31bc28a7fa79acdfa5d1744ce8.

appleseed-iii committed:
Temp messaging for onhold migration (#876)

* temporarily place bond purchases on hold
 - messages per the migrations team
 - can revert this change by finding all the lines that say : NOTE (appleseed): temporary for ONHOLD MIGRATION

* full width

appleseed-iii reviewed (approved):
code looks good. thank you ser. 
@ohmygirth wdyt

appleseed-iii committed:
migration announcement (#884)

appleseed-iii committed:
Revert "Revert "Alpha pr for Migration"" (#883)

* Revert "Revert "Alpha pr for Migration (#869)" (#873)"

This reverts commit 06b303eb8e9498421688653e4917702cf16c3e20.

* prettier fix

appleseed-iii committed:
Staging: Migration for V2 (#881)

* naming conventions, math for sOHM amount, state for wrap/unwrap

* Fix address error

* Fix gohm wrap/unwrap

* Fix wrap v2

* make stake-v2 show up when no assets are detected (not old assets && not new assets)

* clean up migrate modal when no Old Assets Detected & add link to Stak… (#889)

* clean up migrate modal when no Old Assets Detected & add link to Stake V2

* fix btn text size

Co-authored-by: mrunbanked <mrunbanked@protonmail.com>
Co-authored-by: borbesaur <87902542+borbesaur@users.noreply.github.com>

appleseed-iii committed:
should be reverting all of migration off of develop

appleseed-iii committed:
remove 33-t

appleseed-iii committed:
Revert "revert the migration commits (from alpha) off of develop one more time" (#893)

appleseed-iii committed:
fix unstake allowance check, rename ohmStake allowances to ohmStakeV1, & ohmStakeV2 renamed ohmStake, to be consistent with naming

appleseed-iii committed:
typo fix

nabioz committed:
Fix contract interfaces & asset states (depending on user asset combination v1 vs v2) (#894)

* Fix contract interfaces

* Fix typo

* Disable zaps and remove new icon

* state updates
1. add sOHMv2 to `newAssetsDetected` logic
2. add sOHMv2 balance to v1-stake (in case user has it & is waiting on bonds)
3. add `goToBonds` prompt on v1-stake if user has v1 bonds to claim before migration

* update contract addresses

* fix unwrap pending status

* remove old console log

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

appleseed-iii committed:
update wsOHM Price text to gOHM Price (#895)

appleseed-iii committed:
mobile migration styles (#897)

appleseed-iii committed:
update announcement & migration modal styling (#899)

appleseed-iii committed:
seconds to epoch (#900)

* seconds to epoch

* remove dangling comments & logs

appleseed-iii committed:
fix announcement styles (#903)

borbesaur committed:
Unstake -> gohm works (#904)

* Fix contract interfaces

* adding checkbox

* adding ability to unstake from gohm

* add formatting

* stake added and setmax fixed for gohm unstake

Co-authored-by: mrunbanked <mrunbanked@protonmail.com>

appleseed-iii committed:
move dialog (#905)

* move dialog

* move the dialog

borbesaur committed:
Migrate launch bo (#906)

* Fix contract interfaces

* adding checkbox

* adding ability to unstake from gohm

* add formatting

* stake added and setmax fixed for gohm unstake

* approving unstaker gOHM should work

Co-authored-by: mrunbanked <mrunbanked@protonmail.com>

appleseed-iii committed:
update dialog text for gohm stake (#908)

appleseed-iii committed:
Migrate launch bo (#909)

* Fix contract interfaces

* adding checkbox

* adding ability to unstake from gohm

* add formatting

* stake added and setmax fixed for gohm unstake

* approving unstaker gOHM should work

* move checkbox back down

* units formated for staking vs unstaking

* adds message to confirmbox on stake pge

* ginna work for sure

* fix prettier

Co-authored-by: mrunbanked <mrunbanked@protonmail.com>
Co-authored-by: borbersaur <plasmabeach@gmail.com>

borbesaur committed:
Update stake conversion messgare (#910)

* Fix contract interfaces

* adding checkbox

* adding ability to unstake from gohm

* add formatting

* stake added and setmax fixed for gohm unstake

* approving unstaker gOHM should work

* move checkbox back down

* units formated for staking vs unstaking

* adds message to confirmbox on stake pge

* ginna work for sure

* modified staking conversion message for clarity

Co-authored-by: mrunbanked <mrunbanked@protonmail.com>

biwano committed:
Added Polish as a translatable language (No UI yet) (#919)

Fixed Vietnamese flag

Co-authored-by: Bruno Ilponse <bruno.ilponse@gmail.com>

appleseed-iii committed:
FIX: update apy, treasury warning updates, hide migration when not on… (#915)

* FIX: update apy, treasury warning updates, hide migration when not on mainnet

* hide migration modal on TreasuryDashboard, fix small number display from trimmed

* update treasury text

brightiron committed:
wrapping strings with translation tags (#917)

bigint commented on src/hooks/useENS.ts:
```diff
@@ -0,0 +1,31 @@
+const useENS = (address: string) => {
+  const { provider } = useWeb3Context();
+  const [ensName, setENSName] = useState<string | null>(null);
+  const [ensAvatar, setENSAvatar] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
```
@sasicodes I can see `ensAvatar` and `loading` are not yet used, can we remove them or add them in UI?

/cc @appleseed-iii

appleseed-iii replied:
we also have directions in our README for how to resolve merge conflicts in the translations submodule... can you do that? Then let's let this rip.

borbesaur committed:
default staking to sOHM, fix staking conversion text, smaller textbox (#925)

appleseed-iii committed:
fix for ohmv2 - added to newAssetsDetected checks (#924)

* fix for ohmv2

* fix error

appleseed-iii committed:
format strings to bignumber before MAX math on Stake v2 (#930)

appleseed-iii committed:
fix stake button svg border cut-off (#932)

cptnanon committed:
added new cookieFlags (#934)

nabioz committed:
Fix AVAX and Arbitrum Migration UI (#914)

* Save progress

* Remove unused code

* Change button text

* Fix approval

* Remove log

* Change to getBalances

* load account details when chain switch happens

goatsOnDrugs committed:
add check for more that ten dollars of assets for migration cta (#935)

* add check for more that ten dollars of assets for migration cta

* added assetsEnoughToMigrate to the stake or stakev2 turnary

Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>

biwano committed:
Support for languages that are read right to left (#835)

* Support for languages that are read right to left

* fix merge conflict ... :[]

Co-authored-by: bilponse <bilponse@localhost.localdomain>
Co-authored-by: Bruno Ilponse <bruno.ilponse@gmail.com>
Co-authored-by: appleseed-iii <appleseed-iii@protonmail.com>
